src/redux/books/booksSlice.js

```diff
@@ -1,43 +1,56 @@
 import { createSlice } from '@reduxjs/toolkit';
+import unique from '../../util/unique';
+import { fetchBooks, addBook, removeBook } from './booksThunks';
 
 const initialState = {
-  items: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  status: 'idle', // 'idle', 'loading', 'succeeded', 'failed'
+  error: null,
 };
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
-  reducers: {
-    removeBook(state, action) {
-      return {
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      .addCase(fetchBooks.pending, (state) => ({ ...state, status: 'loading' }))
+      .addCase(fetchBooks.fulfilled, (state, action) => ({
         ...state,
-        items: [...state.items.filter((item) => item.item_id !== action.payload)],
-      };
-    },
-    addBook(state, action) {
-      state.items.push(action.payload);
-    },
+        // TODO: handle (should be redundant) render
+        books: unique(action.payload, state.books),
+        status: 'succeeded',
+      }))
+      .addCase(fetchBooks.rejected, (state, action) => ({
+        ...state,
+        error: action.error.message,
+        status: 'failed',
+      }))
+      .addCase(addBook.pending, (state) => ({ ...state, status: 'loading' }))
+      .addCase(addBook.fulfilled, (state, action) => ({
+        ...state,
+        books: [...state.books, action.payload],
+        status: 'succeeded',
+      }))
+      .addCase(addBook.rejected, (state, action) => ({
+        ...state,
+        error: action.error.message,
+        status: 'failed',
+      }))
+      .addCase(removeBook.pending, (state) => ({ ...state, status: 'loading' }))
+      .addCase(removeBook.fulfilled, (state, action) => ({
+        ...state,
+        books: [...state.books.filter((item) => item.item_id !== action.payload)],
+        status: 'succeeded',
+      }))
+      .addCase(removeBook.rejected, (state, action) => ({
+        ...state,
+        error: action.error.message,
+        status: 'failed',
+      }));
   },
 });
-
-export const { removeBook, addBook } = booksSlice.actions;
+export const booksState = (state) => state.books;
+export { fetchBooks, addBook, removeBook };
+// export const { removeBook } = booksSlice.actions;
```

Comment:
- [ ] Kindly make sure you remove all commented and unused code lines from your code base to make it cleaner, professional and for best practices